models/contrib/nish/abracadabra.py

```diff
@@ -0,0 +1,423 @@
+from typing import (
+    Tuple,
+    List,
+)
+from datetime import datetime, timedelta, timezone, date
+from models.tmp_abi_lookup import ABRACADABRA_CAULDRON_ABI
+from credmark.cmf.model import Model
+from credmark.cmf.types import (
+    Address,
+    Contract,
+    Token,
+)
+from credmark.dto import (
+    DTO,
+    EmptyInput,
+    IterableListGenericDTO,
+)
+
+# Function to catch value error in Cauldron v1while fetching mandatory data
+def try_or(func, default=0, expected_exc=(Exception,)):
+    try:
+        return func()
+    except expected_exc:
+        return default
+
+
+# Get token balance of an address on ethereum chain
+def ethereum_token_balance_of_address(contract_address, account_address):
```

Comment:
This whole thing can be replaced with `Token(address=contract_address).scaled(Token(address=contract_address).functions.balanceOf(address))`

Please use the tools taht already exist. if you need ETH you can do the same thing with NativeToken